src/Raven.Server/ServerWide/Maintance/ClusterObserver.cs

```diff
@@ -69,7 +67,10 @@ class ClusterObserver : IDisposable
                     {
                         _logger.Info($"Closing observer on {_nodeTag}, caused by an interrupt.", e);
                     }
-                    return;
+                }
+                finally
+                {
+                    await delay;
```

Comment:
I think we should keep it alive, unless it is gracefully closed with the cancellation token.